### PR TITLE
fix(Views v2) use Tribe__Cache for URL caching, fix keys

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://moderntribe.atlassian.net/browse/TEC-3133  

I found this issue while working on the ticket where View URL caching would lead to the first cached URL (e.g. the non-canonical one) to being always returned for the View.
This would lead to incorrect URL when requesting a view URL w/ diff. parameters from the originally cached one.

### CODE REVIEW

File: src/Tribe/Views/V2/Views/Day_View.php [hunk 1]
```diff
@@ -57,7 +57,53 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
...
+	 * @since 4.9.10
+	 *
+	 * @param mixed $url_date          The date to build the URL for, a \DateTime object, a string or a timestamp.
+	 * @param bool  $canonical         Whether to return the canonical (pretty) version of the URL or not.
+	 * @param array $passthru_vars     An optional array of query variables that should pass thru the method untouched
+	 *                                 in key in value.
+	 *
+	 * @return string The Day View URL for the date.
+	 */
+	protected function build_url_for_date( $url_date, $canonical, array $passthru_vars = [] ) {
```
Comment (borkweb) [inline src/Tribe/Views/V2/Views/Day_View.php]: Moving this method made it hard to tell what (if anything) changed.

Comment (borkweb) [reply src/Tribe/Views/V2/Views/Day_View.php]: Ah, just spacing before `=` on line 84 was the only change.

File: src/Tribe/Views/V2/Views/Day_View.php [hunk 2]
```diff
@@ -222,7 +224,8 @@ protected function setup_messages( array $events ) {
 			$keyword = $this->context->get( 'keyword', false );
 
 			if ( $keyword ) {
-				$this->messages->insert( Messages::TYPE_NOTICE, Messages::for_key( 'no_results_found_w_keyword', trim( $keyword ) ) );
+				$this->messages->insert( Messages::TYPE_NOTICE,
+					Messages::for_key( 'no_results_found_w_keyword', trim( $keyword ) ) );
```
Comment (borkweb) [inline src/Tribe/Views/V2/Views/Day_View.php]: Breaking here doesn't follow our standards. Perhaps:

```php
$this->message->insert(
  Messages::TYPE_NOTICE,
  Messages::for_key( 'no_results_found_w_keyword', trim( $keyword ) )
);
```

File: src/Tribe/Views/V2/Views/Day_View.php [hunk 3]
```diff
@@ -100,66 +147,21 @@ protected function setup_repository_args( \Tribe__Context $context = null ) {
 
 		$context_arr = $context->to_array();
 
-		$date = Arr::get( $context_arr, 'event_date', 'now' );
-		$event_display = Arr::get( $context_arr, 'event_display_mode', Arr::get( $context_arr, 'event_display' ), 'current' );
+		$date          = Arr::get( $context_arr, 'event_date', 'now' );
+		$event_display = Arr::get( $context_arr, 'event_display_mode', Arr::get( $context_arr, 'event_display' ),
+			'current' );
```
Comment (borkweb) [inline src/Tribe/Views/V2/Views/Day_View.php]: Breaking here doesn't match our standards.

Perhaps:

```php
$event_display = Arr::get(
  $context_arr,
  'event_display_mode',
  Arr::get( $context_arr, 'event_display' ),
  'current'
);
```

**Comment by lucatume** [comment]
I will close this PR in favour of another one that contains latest updates.

**Comment by lucatume** [comment]
Closing this in favor of #3024 3024

**Pull request closed without merging**